back/engines/commercial/analysis/app/lib/analysis/summarization_method/gpt4.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'openai'
+
+module Analysis
+  class SummarizationMethod::Gpt4 < SummarizationMethod::Base
+    SUMMARY_TYPE = 'gpt4'
+    GPT_MODEL = 'gpt-4'
+
+    def initialize(*args, **kwargs)
+      super
+      @openai_api = kwargs[:openai_api] || OpenaiApi.new(
+        access_token: ENV.fetch('OPENAI_API_KEY'),
+        request_timeout: 480
+      )
+    end
+
+    protected
+
+    # Use `execute` on the parent class to actually use the method
+    def run
+      summary.update(prompt: prompt)
```

Comment:
`update` can silently fail and therefore I think it would be better to use `update!` instead. Alternatively, we could use `update` and do something with the return value. For example, logging an error to Sentry.

Reply:
:+1:  Turning it into `update!`